snuba/pipeline/stages/query_processing.py

```diff
@@ -0,0 +1,45 @@
+from typing import Any, cast
+
+from snuba.clickhouse.query import Query as ClickhouseQuery
+from snuba.datasets.entities.factory import get_entity
+from snuba.datasets.plans.query_plan import QueryRunner
+from snuba.pipeline.composite import CompositeExecutionPipeline
+from snuba.pipeline.query_pipeline import QueryPipelineData, QueryPipelineStage
+from snuba.query.composite import CompositeQuery
+from snuba.query.data_source.simple import Table
+from snuba.query.logical import Query as LogicalQuery
+from snuba.request import Request
+
+
+class EntityAndStoragePipelineStage(
+    QueryPipelineStage[Request, ClickhouseQuery | CompositeQuery[Table]]
+):
+    def _process_data(
+        self, pipe_input: QueryPipelineData[Request]
+    ) -> ClickhouseQuery | CompositeQuery[Table]:
+        _clickhouse_query = None
+
+        def _query_runner(
+            clickhouse_query: ClickhouseQuery | CompositeQuery[Table],
+            *args: Any,
+            **kwargs: str
+        ) -> None:
+            nonlocal _clickhouse_query
+            _clickhouse_query = clickhouse_query
```

Comment:
This is the key that let's us decouple the execution from the processing. we use the entire old processing pipline but we pass in a dummy runner which just lets us know what the query to run is.